### PR TITLE
add draft prefix, address my own review comment

### PR DESCRIPTION
**Pull request by slingamn**

### CODE REVIEW

File: extensions/chathistory.md [hunk 1]
```diff
@@ -20,13 +28,13 @@ The server as mentioned in this document may refer to either an IRC server or an
 ## Implementation
 The `chathistory` extension uses the [chathistory][batch/chathistory] batch type and introduces a client command, `chathistory`.
 
-To fully support this extension, clients MUST support the [`batch`][batch], [`server-time`][server-time] and [`message-tags`][message-tags] capabilities.
+Full support for this extension requires support for the [`batch`][batch], [`server-time`][server-time] and [`message-tags`][message-tags] capabilities. However, limited functionality is available without support for these CAPs. Servers SHOULD NOT enforce that clients support all related capabilities before using the `chathistory` extension.
```
Comment (prawnsalad) [inline extensions/chathistory.md:31]: What's the rationale behind this change? Having an implementation that doesn't use these only makes it possible to have half completed implementations when we should be pushing for complete implementations. With the issues of deduplication that has been discussed for this spec so far, the lack of say message-tags would make this impossible.

Comment (slingamn) [reply extensions/chathistory.md:31]: @kylef thought that `chathistory` wouldn't be usable as a replacement for `znc.in/playback`, because `znc.in/playback` is usable without support for those caps. However, I think it's very advantageous for `chathistory` to have feature parity with `znc.in/playback` (and in fact you can express any `*playback` message as either a CHATHISTORY AFTER or a CHATHISTORY BETWEEN query and get the same results --- maybe you'll get a duplicated message but it's not the end of the world).

If you see the pseudocode I added here: https://github.com/prawnsalad/ircv3-specifications/pull/2 it offers the possibility of using CHATHISTORY without full deduplication support.

**Comment by slingamn** [comment]
BTW, I added temporary support for `BEFORE *` to Oragono and Kiwi seems to work with it (yay Kiwi!), but I'm not enthusiastic about it as a spec change.

If we're going to replace `LATEST` entirely, we have to replace both `LATEST *` with `BEFORE *` and `LATEST <selector>` with `BETWEEN * <selector>`. This introduces an asymmetry between `BEFORE` and `AFTER`, but it's even more confusing with `BETWEEN`. The semantics of `BETWEEN` with `*` are uniquely problematic because we don't know initially whether we're going to be ascending or descending, we have to figure it out from the order of the selectors. I think any rewrite to accommodate this in `BETWEEN` is ultimately just going to say, "here's this extra special case of `BETWEEN` that has the semantics formerly associated with `LATEST <selector>`". So I think it probably makes sense to keep `LATEST`.

File: extensions/chathistory.md (hunk 1 above)
Comment (slingamn) [reply extensions/chathistory.md:31]: Jess also mentioned this: https://git.causal.agency/pounce/about/pounce.1#Extensions

Basically, right now, the only option is to have the server guess how much history you're missing and play it back. That's one endpoint of a spectrum, and the other endpoint is CHATHISTORY with full deduplication support. But there is an appetite for intermediate points on that spectrum, like `znc.in/playback` and `casual.agency/consumer`, for use by clients that don't want to build the full infrastructure needed to support deduplication. My feeling is that CHATHISTORY will cover those cases if we let it, i.e., by removing the MUST language for dependent caps as done here.

**Comment by slingamn** [comment]
cc @delthas: IMO this is the latest version of the specification.

**Comment by slingamn** [comment]
Thoughts on merging this into the main PR?

**Comment by slingamn** [comment]
I'm considering opening a separate PR on ircv3/ircv3-specifications for this. Thoughts?
